### PR TITLE
Studentsv2

### PR DESCRIPTION
**Pull request by jes708**

### CODE REVIEW

File: browser/js/actions/AuthActions.js [hunk 1]
```diff
@@ -2,21 +2,30 @@
 
 import axios from 'axios'
 
+export const AUTH_LOGIN_REQUEST = 'AUTH_LOGIN_REQUEST'
```
Comment (omriAckley) [inline browser/js/actions/AuthActions.js:5]: Since you're already es6y you might consider using [symbols](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Symbol) instead of strings here. Not a big deal, just seems like the right tool for the job.

File: browser/js/actions/AuthActions.js [hunk 2]
```diff
@@ -35,50 +44,29 @@ export function signup (credentials) {
...
-export function loginSuccess (user) {
-  return {
-    type: AUTH_LOGIN_SUCCESS,
-    id: user.id,
-    user
-  }
-}
-
 export function getLoggedInUser () {
   return (dispatch, getState)  => {
```
Comment (omriAckley) [inline browser/js/actions/AuthActions.js:48]: Couldn't you simply export _this_ function? The outer wrapping function seems irrelevant here.

File: browser/js/actions/AuthActions.js [hunk 3]
```diff
@@ -35,50 +44,29 @@ export function signup (credentials) {
...
     let user = getState().session.user
     if (user) {
-      dispatch(receivedLoggedInUser(user))
+      dispatch(userReceived(user))
```
Comment (omriAckley) [inline browser/js/actions/AuthActions.js:51]: Not sure I'm personally crazy about this event-based architecture I'm seeing, but you all seem to be executing it pretty well and establishing solid patterns.

File: browser/js/actions/userAssessmentActions.js [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+
+const API_URL = '/api/v1/assessments'
+
+export function getUserAssessments (id) {
+  return dispatch => {
+    dispatch({ type: LOAD_ASSESSMENTS_REQUEST })
+    return axios.get(API_URL)
+    .then(res => res.data)
+    .then(resData => {
+      console.log('resData', resData);
```
Comment (omriAckley) [inline browser/js/actions/userAssessmentActions.js:17]: Make sure to keep unnecessary or dead code out of master :)

File: browser/js/components/GraderAssessments.js [hunk 5]
```diff
@@ -1,12 +1,15 @@
...
 import Paper from 'material-ui/Paper';
 import RaisedButton from 'material-ui/RaisedButton';
 import FlatButton from 'material-ui/FlatButton';
 import FontIcon from 'material-ui/FontIcon';
+import Chip from 'material-ui/Chip'
 import { Card, CardActions, CardHeader, CardText } from 'material-ui/Card';
 import { RadioButton, RadioButtonGroup } from 'material-ui/RadioButton';
+import { getUserAssessments } from '../actions/userAssessmentActions'
 
 const styles = {
```
Comment (omriAckley) [inline browser/js/components/GraderAssessments.js:14]: Consider moving this to another file. Reason one: this current file is getting a little long. Reason two: these styles are likely to just keep growing. Reason three: it seems like you might want to be able to work on the styles and the component logic both independently and concurrently, so having them in different files can help you avoid future unnecessary merge conflicts.

File: browser/js/components/GraderAssessments.js [hunk 6]
```diff
@@ -44,39 +47,71 @@ const styles = {
...
+                {this.renderTags(assessment.tags)}
+              </div>
             </div>
           </Card>
+        )
+      })
+    }
+  }
+
+  render () {
```
Comment (omriAckley) [inline browser/js/components/GraderAssessments.js:90]: Digging the modularity you've got going on.

File: browser/js/components/GraderStudents.js [hunk 7]
```diff
@@ -60,24 +67,52 @@ const SAMPLE_SPEC = {
...
   }
-  render () {
-    return (
-      <div style={Object.assign(styles.gradingPane, styles.paperStyle)}>
-        <div style={styles.content}>
-          <Card style={Object.assign(styles.infoCard, styles.skinny)}>
+
+  renderStudents () {
+    if (!this.props.teamFetching && this.props.team) {
+      console.log("props", this.props)
```
Comment (omriAckley) [inline browser/js/components/GraderStudents.js:78]: Channel your inner lumberjack

File: browser/js/config/configureStore.js [hunk 8]
```diff
@@ -8,12 +8,13 @@ import rootReducer from '../reducers';
 const loggerMiddleware = createLogger();
 
 export default function configureStore (initialState) {
+  const middlewares = [thunkMiddleware]
+  if (process.env.NODE_ENV !== 'production') {
```
Comment (omriAckley) [inline browser/js/config/configureStore.js:12]: Is this working? This file seems like it's going to get executed in the browser context and would not have access to `process`.

File: browser/js/reducers/assessmentTeam.js [hunk 9]
```diff
@@ -0,0 +1,34 @@
...
+  LOAD_TEAM_FAILURE
+} from '../actions/assessmentTeamActions'
+
+const initialState = {
+  teamFetching: false,
+  error: null
+}
+
+export default function assessmentTeam (state = initialState, action) {
+  console.log('assessmentTeam.js', action)
```
Comment (omriAckley) [inline browser/js/reducers/assessmentTeam.js:15]: Disposable.

File: browser/js/reducers/assessmentTeam.js [hunk 10]
```diff
@@ -0,0 +1,34 @@
...
+    case LOAD_TEAM_REQUEST:
+      return Object.assign({}, state, {
+        teamFetching: true
+      })
+    case LOAD_TEAM_SUCCESS:
+      return Object.assign({}, state, {
+        teamFetching: false,
+        team: action.team
+      })
+    case LOAD_TEAM_FAILURE:
```
Comment (omriAckley) [inline browser/js/reducers/assessmentTeam.js:26]: Not sure if this would be the right spot for it, but you might consider implementing a simple error displaying/reporting system for errors that are important to the end user. For example there's [this library](https://github.com/Moosylvania/react-growl) which seems cool (full disclosure: I've used stuff like this, but not this exact library).

File: browser/js/reducers/userAssessments.js [hunk 11]
```diff
@@ -0,0 +1,35 @@
...
+} from '../actions/userAssessmentActions'
+
+const initialState = {
+  isFetching: false,
+  items: [],
+  error: null
+}
+
+export default function userAssessments (state = initialState, action) {
+  console.log('userAssessments.js', action)
```
Comment (omriAckley) [inline browser/js/reducers/userAssessments.js:16]: Erase

File: server/app/routes/api/v1/teams/index.js [hunk 12]
```diff
@@ -19,11 +19,42 @@ const models = db()
...
 const Resource = Team;
 
 /** see documentation at https://www.npmjs.com/package/sequelize-handlers */
+
+router.use(function(req, res, next) {
+  req.options = {
+        where: {
+            id: req.params.id,
+        }, 
+        include: [
```
Comment (omriAckley) [inline server/app/routes/api/v1/teams/index.js:34]: You might also achieve a similar goal using [sequelize scope](http://docs.sequelizejs.com/en/latest/docs/scopes/). And in my opinion, the more logic you can shift to the model level, the better.

File: server/db/models/Organizations/Team/methods.js [hunk 13]
```diff
@@ -22,7 +22,7 @@ function addAssociations(db){
   const UserTeam = db.models['userTeam'];
```
Comment (omriAckley) [inline server/db/models/Organizations/Team/methods.js:22]: Could just be `db.model.userTeam`.

**Comment by omriAckley** [comment]
Overall this looks fantastic. I think you're doing a good job not only writing modular code but also establishing reusable patterns across your application.

File: server/app/routes/api/v1/teams/index.js [hunk 14]
```diff
@@ -19,11 +19,42 @@ const models = db()
...
+                role: 'instructor'
+              }
+            }
+          }
+        ]
+    }
+  next();
+})
+
 router.get( '/', sequelizeHandlers.query( Resource ) );
```
Comment (omriAckley) [inline server/app/routes/api/v1/teams/index.js:58]: This `sequelizeHandlers` utility seems slick!

File: browser/js/components/GraderStudents.js (hunk 7 above)
Comment (jdhang) [inline browser/js/components/GraderStudents.js:78]: Love this ^

File: browser/js/config/configureStore.js (hunk 8 above)
Comment (jdhang) [inline browser/js/config/configureStore.js:12]: When Webpack builds our code for production it will address this correctly, I tested it by telling webpack to build for production and there were no log messages other than our missed console.logs(). We just had to add an extra webpack plugin into our webpack config file luckily.
